assets/app/scripts/services/applicationGenerator.js

```diff
@@ -88,7 +88,15 @@ angular.module("openshiftConsole")
       var service = scope._generateService(input, input.name, ports);
       if (service) {
         resources.service = service;
+
         // Only attempt to generate a route if there is a service.
+        // Use the port name from the service when generating the route.
+        if (input.routing.targetPort) {
+          var servicePort = _.find(service.spec.ports, function(servicePort) {
```

Comment:
It shouldn't since we generate a service port for any container port, but I guess the input from the caller could be bad. I'll add a null check.